spark/v3.5/spark-runtime/NOTICE

```diff
@@ -500,9 +363,312 @@ file:
 This binary artifact includes Project Nessie with the following in its NOTICE
 file:
 
-| Dremio
-| Copyright 2015-2017 Dremio Corporation
-|
-| This product includes software developed at
-| The Apache Software Foundation (http://www.apache.org/).
-
+| Nessie
+| Copyright 2015-2025 Dremio Corporation
+| 
+| ---------------------------------------
+| This project includes code from Apache Iceberg, with the following in its NOTICE file:
+| 
+| | Apache Iceberg
+| | Copyright 2017-2025 The Apache Software Foundation
+| |
+| | This product includes software developed at
+| | The Apache Software Foundation (http://www.apache.org/).
+| |
+| | --------------------------------------------------------------------------------
+| |
+| | This project includes code from Kite, developed at Cloudera, Inc. with
+| | the following copyright notice:
+| |
+| | | Copyright 2013 Cloudera Inc.
+| | |
+| | | Licensed under the Apache License, Version 2.0 (the "License");
+| | | you may not use this file except in compliance with the License.
+| | | You may obtain a copy of the License at
+| | |
+| | |   http://www.apache.org/licenses/LICENSE-2.0
+| | |
+| | | Unless required by applicable law or agreed to in writing, software
+| | | distributed under the License is distributed on an "AS IS" BASIS,
+| | | WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+| | | See the License for the specific language governing permissions and
+| | | limitations under the License.
+| 
+| ---------------------------------------
+| This project includes code from Apache Polaris (incubating), with the following in its NOTICE file:
+| 
+| | Apache Polaris (incubating)
+| | Copyright 2024 The Apache Software Foundation
+| |
+| | This product includes software developed at
+| | The Apache Software Foundation (http://www.apache.org/).
+| |
+| | The initial code for the Polaris project was donated
+| | to the ASF by Snowflake Inc. (https://www.snowflake.com/) copyright 2024.
+| 
+| ---------------------------------------
+| This project includes code from Netty, with the following in its NOTICE file:
```

Comment:
It's too bad that this was based on a source inclusion in Nessie. Nessie could probably check that none of this applies to ResolveConf that was copied to avoid more work for downstream bundlers.

Reply:
I think Nessie is using a similar approach as in Iceberg: as discussed, Iceberg is using `LICENSE` and `NOTICE` from source in jar files by default (only bundle and runtime jars have specific `LICENSE` and `NOTICE`.
For instance, Iceberg mentions `ScriptRunner` which is used for iceberg-hive-metastore test, but mention in all Iceberg jar. Or `AssignmentAlignmentSupport` is used only in iceberg-spark-extensions but mentioned in all iceberg jar.
So, I'm not very confortable to ask Nessie to change that (even if it's what I asked but the Nessie team says they will plan this as it's more work) when Iceberg is doing the same.

Reply:
The difference is that we're more selective with NOTICE content when code is copied. That's because we can more easily verify that sections of the original project's NOTICE do or do not apply to the specific section that was copied or used.